server/src/main/java/org/elasticsearch/cluster/routing/allocation/RoutingAllocation.java

```diff
@@ -143,6 +139,20 @@ private RoutingAllocation(
         this.nodeReplacementTargets = nodeReplacementTargets(clusterState);
         this.desiredNodes = DesiredNodes.latestFromClusterState(clusterState);
         this.unaccountedSearchableSnapshotSizes = unaccountedSearchableSnapshotSizes(clusterState, clusterInfo);
+        this.routingChangesObserver = new RoutingChangesObserver.DelegatingRoutingChangesObserver(
+            isSimulating
```

Comment:
Is there any particular reason we don't want log shard changes in the non-simulating mode?

Reply:
Yeap, simulating is a mode when we compute the desired balance allocation, not preform a real shard movements, so I do not think they are relevant for the user.

Here we would like to record actual shard and data movements that happened in the cluster.